db/users.go

```diff
@@ -121,6 +121,15 @@ func (dbc *DatabaseContext) UpdatePrincipal(ctx context.Context, updates *auth.P
 			changed = true
 		}
 
+		for scopeName, collections := range updates.CollectionAccess {
```

Comment:
To avoid an additional over collections, do this check in `dbc.RequiresCollectionAccessUpdate` and change that to return err as well. We can exit early there.